lib/internal/Magento/Framework/Filesystem/DirectoryList.php

```diff
@@ -97,6 +97,10 @@ public function __construct($root, array $config = [])
         $this->root = $this->normalizePath($root);
         $this->directories = static::getDefaultConfig();
         $this->directories[self::SYS_TMP] = [self::PATH => realpath(sys_get_temp_dir())];
```

Comment:
Hi @nikolaevas. Thank for collaboration. 
You can use ternary operator in this case.
Something like: 
$this->directories[self::SYS_TMP] ? ini_get('upload_tmp_dir') ? [self::PATH => realpath($uploadTmpDir)] : [self::PATH => realpath(sys_get_temp_dir())]

Reply:
Hi @VladimirZaets. I have updated the code.